### ISSUE TITLE
[12.0] web_widget_one2many_product_picker, limit initial loaded products by domain

### ISSUE THREAD

**Issue by GSLabIt** [OCA/web#2039]
Hi there, 

is there any way to limit initial product loaded by a domain? Let's say we'd like to load only service products in the widget, how this can be done? 

We've been able to load different product per groups, but not limiting initial loaded products. 

On v `13.0`, `all_domain` is available, but it still load all products without any domain at initial load but, if group is selected it apply the domain.

Basically we'd like to have a domain to be applied in `all` group for initial loaded products. 

@Tardo @pedrobaeza,  any hint on how to do?

**Comment by pedrobaeza** [comment]
@Tardo, isn't `all_domain` initially applied without such glitch?

**Comment by GSLabIt** [comment]
@pedrobaeza @tafaRU i think i've been able to manage it. See https://github.com/OCA/web/pull/2040, review super welcome.

**Comment by Tardo** [comment]
When you say "all groups" you say "all groups" or the "all" group hehehe... ??

With the changes in the PR that you have proposed the way to have the same domain for all the groups is to put that domain manually in each group.
You could expand the options and modify "_getSearchRecords" to inject a generic domain for all groups.

in v13 "all_domain" should be applied in the initial load :/ ... thanks for report.

**Comment by GSLabIt** [comment]
> When you say "all groups" you say "all groups" or the "all" group hehehe... ??
> 
> With the changes in the PR that you have proposed the way to have the same domain for all the groups is to put that domain manually in each group.
> You could expand the options and modify "_getSearchRecords" to inject a generic domain for all groups.
> 
> in v13 "all_domain" should be applied in the initial load :/ ... thanks for report.

I mean del `all` group; what i propose in the pr is the same feature in v. 13 so that the `all_domain` will be applied to the initial load.

I've tested loacally and it seems to work.

**Comment by Tardo** [comment]
Maybe can be interesting add an option to disable "All" group?

**Comment by pedrobaeza** [comment]
> Maybe can be interesting add an option to disable "All" group?

I don't think so, as a group at least should exist. The only thing can be to be able to rename it.

Closing as the main issue is solved.
